Commands/process.go

```diff
@@ -0,0 +1,54 @@
+package Commands
+
+import (
+	"errors"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/slack-go/slack"
+)
+
+func ProcessMessage(channel string, msgTimestamp string, msg string) error {
+
+	botID := "<@U025MPQCB9A> "
+
+	name, err := parseCommandName(strings.TrimPrefix(msg, botID))
+	if err != nil {
+		return err
+	}
+
+	response, err := RunCommand(name)
+	replyWith(channel, msgTimestamp, response)
+
+	return err
+}
+
+func parseCommandName(msg string) (string, error) {
+	commands := map[string]string{
+		"balance":  `^my balance`,
+		"transfer": `^transfer to`,
+	}
+	for name, pattern := range commands {
+		matched, err := regexp.MatchString(pattern, msg)
+		if err != nil {
+			return "", nil
+		}
+		if matched {
+			return name, nil
+		}
+	}
+	return "", errors.New("no match")
+}
+
+func replyWith(channel string, msgTimestamp string, response string) (err error) {
+	botSecret := os.Getenv("SLACK_BOT_SECRET")
+	var api = slack.New(botSecret)
+
+	api.PostMessage(
+		channel,
+		slack.MsgOptionText(response, false),
+		slack.MsgOptionTS(msgTimestamp), // reply in thread
+	)
+	return err
```

Comment:
`err` here is always `nil` because it never gets set to any value in the body of the function. In general, avoid named return values and prefer to be explicit. They're useful in some specific cases, but it's better to get the error result in `api.PostMessage` and return it to avoid surprises like this.

Reply:
Good one. Will change!